wpiutil/src/main/java/edu/wpi/first/util/sendable2/SendableTable.java

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.util.sendable2;
+
+import edu.wpi.first.util.function.BooleanConsumer;
+import edu.wpi.first.util.function.FloatConsumer;
+import edu.wpi.first.util.function.FloatSupplier;
+import edu.wpi.first.util.struct.Struct;
+
+import java.util.function.BooleanSupplier;
+import java.util.function.Consumer;
+import java.util.function.DoubleConsumer;
+import java.util.function.DoubleSupplier;
+import java.util.function.LongConsumer;
+import java.util.function.LongSupplier;
+import java.util.function.Supplier;
+
+/** Helper class for building Sendable dashboard representations. */
+public interface SendableTable extends AutoCloseable {
+  public static class PubSubOption {
+    // TODO: Move ntcore PubSubOption to wpiutil?
+  }
+
+  void setBoolean(String name, boolean value);
+
+  void publishBoolean(String name, BooleanSupplier supplier, PubSubOption... options);
+
+  BooleanConsumer publishBoolean(String name, PubSubOption... options);
+
+  void subscribeBoolean(String name, BooleanConsumer consumer, PubSubOption... options);
+
+  void setInt(String name, long value);
+
+  void publishInt(String name, LongSupplier supplier, PubSubOption... options);
+
+  LongConsumer publishInt(String name, PubSubOption... options);
+
+  void subscribeInt(String name, LongConsumer consumer, PubSubOption... options);
+
+  void setFloat(String name, float value);
+
+  void publishFloat(String name, FloatSupplier supplier, PubSubOption... options);
+
+  FloatConsumer publishFloat(String name, PubSubOption... options);
+
+  void subscribeFloat(String name, FloatConsumer consumer, PubSubOption... options);
+
+  void setDouble(String name, double value);
+
+  void publishDouble(String name, DoubleSupplier supplier, PubSubOption... options);
+
+  DoubleConsumer publishDouble(String name, PubSubOption... options);
```

Comment:
There are a few styles of overloads here; can you explain what the expected use case / syntax for each is?

Reply:
We might not want these to be overloads, particularly with the variable arguments.  The first one works like today's Sendable property getters, in that the supplier is called periodically.  The second one works like a normal DoublePublisher (and actually will be one under the hood!), in that there is no polling--instead the class can decide when to publish a new value by calling the returned DoubleConsumer.

Reply:
Called periodically by what? 
How will the `DoublePublisher` be closed?
Is `setDouble` analogous to the current `publishConst`?

Reply:
Just like today's SendableBuilderImpl in wpilibj, there will be an implementation class that runs as part of the robot periodic loop that loops over all the SendableTables.

The publishers and subscribers are owned by the SendableTable implementation.

Yes, setDouble is the same as today's publishConst.